### ISSUE TITLE
[BUG] Single Child in ContentScrollSnapHorizontal

### ISSUE THREAD

**Issue by SimonDarksideJ** [Unity-UI-Extensions/com.unity.uiextensions#539]
<sup>Issue created by James Davies as [Bitbucket Issue #&#8203;362](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362) on 2020.12.22 08:27.</sup>
When using the ContentScrollSnapHorizontal from the context menu, and only having 1 item in the content \(removing 4 of the 5 added\) causes the panel not to snap correctly.

**Steps:**

1. Create new scene
2. Right click Hierarchy
3. Choose UI > Extensions > Content Scroll Snap Horizontal
4. Delete Position 2 through to Position 5, so only 1 item exists
5. Run
6. Drag the item

‌

**Comment by SimonDarksideJ** [comment]
<sup>On [2020.12.23 07:33](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-59593881), James Davies commented:</sup>
Managed to fix this issue using the following code:

```csharp
        public void OnBeginDrag(PointerEventData ped)
        {
            if (contentPositions.Count <= 1)
            {
                return;
            }

            StopMovement();
            if (!Moving)
            {
                MovementStarted.Invoke();
            }
        }

        public void OnEndDrag(PointerEventData ped)
        {
            if (contentPositions.Count <= 1)
            {
                return;
            }

            StartCoroutine("SlideAndLerp");
        }
```

All I do is make sure there’s more than 1 item to call the code.

It might not be required to have the filter in the BeginDrag, but if there’s going to be no movement, maybe it shouldn’t call the callback.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.01.26 10:12](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-59817541), @SimonDarksideJ commented:</sup>
Thanks for the feedback @{5b892e2fb2b15c2bdfcc9e88} and for the potential fix.  I’ll investigate further.

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:21](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-60494099), @SimonDarksideJ commented:</sup>
Apologies for the delay, resolved in 2.2.5

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:22](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-60494111), @SimonDarksideJ commented:</sup>
Resolved issue where the Content Scroll snap would cause issue with only 1 child.
Also tested and fixed working with no children.

Resolves #362


Referenced by baca543db780

**Comment by SimonDarksideJ** [comment]
<sup>On 2021.05.10 15:22 @SimonDarksideJ modified issue:</sup>
**status** changed `new` → `resolved`

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 15:22](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-60494112), @SimonDarksideJ commented:</sup>
Fixed in 2.2.5

**Comment by SimonDarksideJ** [comment]
<sup>On [2021.05.10 18:38](https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/issues/362#comment-60495268), @SimonDarksideJ commented:</sup>
Merged in development (pull request #122)

Development

* Merged in updated_new_input_system_check (pull request #117)

Updated the new Input system switch and tested against 2021

* Updated LineRenderer to initialise with a 1 point array, this resolves an issue with Unity putting the previous controls vertext array in an uninitialised control.
Also updated examples checkout

* Merged in UpdatedLineRender (pull request #118)

UILineRender now initilises with a point of 1 to correct Unity issue

* Applied J.R. Mitchell's fix for the Accordion Controls/Accordion/AccordionElement.cs

resolves: #364

* Merged in AccordionFIx (pull request #119)

Update to Accordion to allow for prefab Accordion Elements

* Resolved issue where the Content Scroll snap would cause issue with only 1 child.
Also tested and fixed working with no children.

Resolves #362

* Merged in content_scroll_snap_fix (pull request #120)

ContentScrollSnap update

* Updated the PaginationManager to override if the ScrollSnap is in motion.
BuyerBeware, Unity's toggles can get in to a state where they are clicked but not clicked.

* Merged in PaginationFix (pull request #121)

Updated the PaginationManager to override if the ScrollSnap is in motion.

* Updated release notes and bumped version to release


Referenced by 383e19fb0512
